GP3.Client/Models/APIErrorParser.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Firebase.Auth;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GP3.Client.Models
+{
+    public static class APIErrorParser
+    {
+        public static string ParseErrorToString(FirebaseAuthException ex)
+        {
+
+            string errorText = "";
+
+            switch (ex.Reason)
+            {
+                case AuthErrorReason.InvalidEmailAddress:
+                    errorText = "Email address and/or password is invalid";
+                    break;
+
+                case AuthErrorReason.UserNotFound:
+                    errorText = "This user does not exist";
+                    break;
+
+                default:
+                    errorText = ex.Message;
+                    break;
+            }
```

Comment:
IMO this is bad.
User shouldn't know if login failed due to bad email, user not existing or password being invalid. Just return that login failed and that should be OK.
(Reason being that they can then logically brute force which emails are registered to firebase)
(Also, I believe this is missing handler for invalid password)